src/main/java/com/opentable/kafka/util/OffsetMonitor.java

```diff
@@ -0,0 +1,137 @@
+package com.opentable.kafka.util;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serdes;
+
+import kafka.admin.AdminClient;
+import scala.collection.JavaConverters;
+
+public class OffsetMonitor implements AutoCloseable, Closeable {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+
+    private final String consumerGroupId;
+    private final KafkaConsumer<byte[], byte[]> consumer;
+    private final AdminClient adminClient;
+
+    /**
+     * Namespace the consumer created by this class; for example, pass your application name.
+     */
+    public OffsetMonitor(final String consumerNamespace, final String brokerList) {
+        consumerGroupId = consumerNamespace + "-offset-monitor";
+        consumer = makeConsumer(brokerList);
+        adminClient = makeAdminClient(brokerList);
+    }
+
+    private KafkaConsumer<byte[], byte[]>  makeConsumer(final String brokerList) {
+        final Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, consumerGroupId);
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, (int) TIMEOUT.toMillis());
+        final Deserializer<byte[]> deser = Serdes.ByteArray().deserializer();
+        return new KafkaConsumer<>(props, deser, deser);
+    }
+
+    private AdminClient makeAdminClient(final String brokerList) {
+        final Properties props = new Properties();
+        props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+        return AdminClient.create(props);
+    }
+
+    @Override
+    public void close() {
+        consumer.close();
+        adminClient.close();
+    }
+
+    /**
+     * Find size (maximum offset) for each partition on topic.
+     *
+     * <p>
+     * If the topic does not exist, the returned map will be empty.
+     *
+     * @param topic the topic for which to fetch partition sizes (maximum offsets)
+     * @return map of partition of size (maximum offset)
+     */
+    public Map<Integer, Long> getTopicSizes(final String topic) {
```

Comment:
Partitions are guaranteed to be consecutive, right?  Would a `long[]` be slightly more readable?  The stream code isn't bad at all but this might be one of those cases where the old-fashioned for loop with an index is more understandable.  It'd also avoid a lot of boxing -- this isn't a performance critical section, so it's a minor note at most.

Reply:
I'll play with this soon.